pkg/controllers/interruption/controller.go

```diff
@@ -144,9 +144,11 @@ func (c *Controller) handleMessage(ctx context.Context, machineInstanceIDMap map
 	nodeInstanceIDMap map[string]*v1.Node, msg messages.Message) (err error) {
 
 	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("messageKind", msg.Kind()))
-	receivedMessages.WithLabelValues(string(msg.Kind())).Inc()
 
 	if msg.Kind() == messages.NoOpKind {
+		receivedMessages.With(prometheus.Labels{
```

Comment:
I'm not sure that anything needs to be changed in how we are currently handling the `receivedMessages` metric. This metric would fire more than needed now with the changes you made. I think we can probably ignore this metric for this set of changes.